rfcs/0002_amethyst_as_bin.md

```diff
@@ -0,0 +1,152 @@
+  in the template. If you don't wish to use the Amethyst input system you can return `None` here instead.
+  
+- `fn game_data() -> impl DataInit<GameData>` 
+  
+  Provides the core of your game's parallelism.  By default we are returning a `GameDataBuilder` type from here with a few bundles
+  added that you will probably find useful.  You can read more about custom GameData types in
+  [The Book](https://www.amethyst.rs/book/latest/). You do not need to provide input or rendering facilities here, as the Amethyst
+  core dispatcher will take care of that for you.
+  
+- `fn pipeline_builder(pipeline: PipelineBuilder<Queue<()>>) -> Option<impl PipelineBuild>`
+  
+  Provides a description of the rendering pipeline.  It is pre-populated with a basic configuration that should
+  work for most needs.  If you don't want a window for your game (such as for a headless game server) then you can return `None` here.
+  
+- `fn post_exit()`
+
+  This function will be executed after your game quits, and the world has been unloaded.  Most games won't need to do anything here but 
+  if you, for example need to trigger an upload to a cloud save after the game is over this is the place to do it.
+
+# Reference-Level Explanation
+[reference-level-explanation]: #reference-level-explanation
+
+During compilation (triggered via a hypothetical cli command `amethyst run`) the amethyst engine links an engine binary against a user 
+provided crate which is expected to provide the interface detailed above. The new main method we provide calls these functions at 
+appropriate times during initialization of the game and/or editor so as to place the user provided values into the correct locations 
+during the startup process.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Since this is all very non-conventional we'll have to build, maintain, and support our own tooling for this workflow.
+Additionally we will have to provide clear and comprehensive learning materials on how to use said tooling.
+
+# Rationale and Alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The primary reason to consider this workflow is that by taking back `fn main()` we can build out features such as an editor and 
+scripting in a way that doesn't require our users to be well versed on how to interface their game with our editor.  With this approach
+we can provide a simple "launch in editor" button and/or command rather than asking users to alter their code to launch in our editor.
+
+The major alternative is to leave things as they are.  While it's a little easier to maintain this way, we're somewhat limited in
+what we can do as a lib. Several people have commented on the issue tracker before "libs shouldn't do this" or "libs shouldn't do that" 
+in response to us doing things that are very reasonable for a game engine to need to do. So my takeaway is that we shouldn't be a lib.
```

Comment:
> that are very reasonable for a game engine to do

Well, that depends a lot on what you expect a game engine to do :smile: It also depends on if you prefer facilities that are provided or frameworks. The latter imposes more structure on your code. This can have advantages sometimes (think of the ECS pattern), but it can also feel very restricting.